scripts/mongodbintegrationmvp.py

```diff
@@ -0,0 +1,230 @@
+parameters = [
+    {"name": "x1", "type": "range", "bounds": [-5.0, 10.0]},
+    {"name": "x2", "type": "range", "bounds": [0.0, 10.0]},
+]
+objectives = {obj1_name: ObjectiveProperties(minimize=True)}
+
+# Use Ax's default Sobol trials for 2D problems
+SOBOL_TRIALS = 5
+
+
+def save_ax_snapshot_to_mongodb(ax_client, experiment_name):
+    """Save Ax client snapshot to MongoDB with timestamp (append, don't overwrite)."""
+    try:
+        temp_file = f"temp_{experiment_name}_snapshot.json"
+        ax_client.save_to_json_file(temp_file)
+
+        with open(temp_file, "r") as f:
+            snapshot_data = json.load(f)
+
+        snapshot_doc = {
+            "experiment_name": experiment_name,
+            "snapshot_data": snapshot_data,
+            "timestamp": datetime.now().isoformat(),
+            "trial_count": (
+                len(ax_client.get_trials_data_frame())
+                if ax_client.get_trials_data_frame() is not None
+                else 0
+            ),
+        }
+
+        # Insert a new document for every snapshot (no overwrite)
+        snapshots_col.insert_one(snapshot_doc)
+
+        os.remove(temp_file)
+
+        print(f"Snapshot saved to MongoDB at {snapshot_doc['timestamp']}")
+        return True
+
+    except Exception as e:
+        print(f"Error saving snapshot: {e}")
+        return False
+
+
+def load_ax_snapshot_from_mongodb(experiment_name):
+    """Load most recent Ax client snapshot from MongoDB."""
+    try:
+        # Find the most recent snapshot
+        record = snapshots_col.find_one(
+            {"experiment_name": experiment_name},
+            sort=[("timestamp", -1)],  # Most recent first
+        )
+
+        if record:
+            # Save snapshot data to temporary file
+            temp_file = f"temp_{experiment_name}_snapshot.json"
+            with open(temp_file, "w") as f:
+                json.dump(record["snapshot_data"], f)
+
+            # Load AxClient from file
+            ax_client = AxClient.load_from_json_file(temp_file)
+
+            # Clean up temp file
+            os.remove(temp_file)
+
+            print(
+                f"Loaded snapshot from {record['timestamp']} with "
+                f"{record['trial_count']} trials"
+            )
+            return ax_client
+        else:
+            print("No existing snapshot found")
+            return None
+
+    except Exception as e:
+        print(f"Error loading snapshot: {e}")
+        return None
+
+
+# Load existing experiment or create new one
+ax_client = load_ax_snapshot_from_mongodb(obj1_name)
+
+if ax_client is None:
+    # Create new experiment
+    generation_strategy = GenerationStrategy(
+        [
+            GenerationStep(
+                model=Models.SOBOL,
+                num_trials=SOBOL_TRIALS,
+                min_trials_observed=1,
+                max_parallelism=5,
+                model_kwargs={"seed": 999},  # For reproducibility
+            ),
+            GenerationStep(
```

Comment:
Since you're using snapshots, probably ok to leave to defaults and not specify an explicit generation strategy